File: VMO-Frontend/src/app/model/Dto/SalaryProfileDto.ts
import { ActiveStatus } from 'src/share/model/ActiveStatus';
import { Benefit } from '../Benefit';
import { Allowance } from '../Allowance';
import { Employee } from '../Employee';
import { StatusSign } from 'src/share/model/StatusSign';


export interface SalaryProfileDto {
  salaryProfileId: string;
  salaryProfileCode: string;
  basicSalary: number;
  bonus: number;
  deduction: number;
  netSalary: number;
  grossSalary: number;
  salaryRank: string;
  salaryLevel: string;
  createDate: Date;
  creatorId?: string;
  updateDate?: Date;
  updaterId?: string;
  signDate: Date;
  statusSign: StatusSign;
  status: ActiveStatus;

  contractId?: string;
  employeeId: string;
  benefits?: Benefit[];
  allowances?: Allowance[];
}

export interface AddSalaryProfileDto {
  basicSalary: number;
  bonus: number;
  deduction: number;
  netSalary: number;
  grossSalary: number;
  salaryRank: string;
  salaryLevel: string;
  createDate: Date;
  creatorId?: string;
  updateDate?: Date;
  updaterId?: string;
  status: ActiveStatus;

  contractId?: string;
  employeeId: string;
  benefits?: Benefit[];
  allowances?: Allowance[];
}

File: VMO-Back/Model/DTO/SalaryProfileDto.cs
using Model.Model;
using Model.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model.DTO
{
    public class SalaryProfileDto
    {
        public string SalaryProfileId { get; set; }
        public int BasicSalary { get; set; }
        public int Bonus { get; set; }
        public int Deduction { get; set; }
        public int NetSalary { get; set; }
        public string SalaryRank { get; set; }
        public string SalaryLevel { get; set; }
        public DateTime CreateDate { get; set; }
        public string CreatorId { get; set; }
        public DateTime UpdateDate { get; set; }
        public string UpdaterId { get; set; }
        public ActiveStatus Status { get; set; }

        //Relationship
        public string? ContractId { get; set; }
        public string EmployeeId{ get; set; }
        public ICollection<Benefit>? Benefits { get; set; }
        public ICollection<Allowance>? Allowances { get; set; }
    }
    public class SalaryProfileAddDto
    {
        public int BasicSalary { get; set; }
        public int Bonus { get; set; }
        public int Deduction { get; set; }
        public int NetSalary { get; set; }
        public string SalaryRank { get; set; }
        public string SalaryLevel { get; set; }
        public DateTime CreateDate { get; set; }
        public string CreatorId { get; set; }
        public DateTime UpdateDate { get; set; }
        public string UpdaterId { get; set; }
        public ActiveStatus Status { get; set; }
        public string? ContractId { get; set; }
        public string EmployeeId { get; set; }

        public ICollection<BenefitDto>? Benefits { get; set; }
        public ICollection<AllowanceDto>? Allowances { get; set; }
    }

    public class SalaryProfileUpdateDto
    {
        public int BasicSalary { get; set; }
        public int Bonus { get; set; }
        public int Deduction { get; set; }
        public int NetSalary { get; set; }
        public string SalaryRank { get; set; }
        public string SalaryLevel { get; set; }
        public DateTime CreateDate { get; set; }
        public string CreatorId { get; set; }
        public DateTime UpdateDate { get; set; }
        public string UpdaterId { get; set; }
        public ActiveStatus Status { get; set; }
        public string? ContractId { get; set; }
        public string EmployeeId { get; set; }

        public ICollection<BenefitSalaryProfile> BenefitSalaryProfiles { get; set; }
        public ICollection<AllowanceSalaryProfile> AllowanceSalaryProfiles { get; set; }
    }
}
